CMakeLists.txt

```diff
@@ -109,6 +111,10 @@ if (BUILD_TESTS AND GTEST_FOUND)
   add_subdirectory(tests)
 endif()
 
+if (RUN_TESTS)
+  set(BUILD_TESTS ON)
```

Comment:
You don't need this. option() should set variable to OFF or to whatever you have specified manually.